src/main/java/teammates/ui/webapi/action/SearchInstructorsAction.java

```diff
@@ -30,9 +33,28 @@ public void checkSpecificAccessControl() {
     public ActionResult execute() {
         String searchKey = getNonNullRequestParamValue(Const.ParamsNames.ADMIN_SEARCH_KEY);
         List<InstructorAttributes> instructors = logic.searchInstructorsInWholeSystem(searchKey).instructorList;
-        InstructorsData instructorsData = new InstructorsData(instructors);
-        // Hide information
+
+        List<InstructorData> instructorDataList = new ArrayList<>();
+        for (InstructorAttributes instructor : instructors) {
+            InstructorData instructorData = new InstructorData(instructor);
+            // Only admin will be able to access this page, so we can go ahead and set the key
+            instructorData.setKey(StringHelper.encrypt(instructor.getKey()));
```

Comment:
Can we standardize one the information hiding? Either we want to do it in this way or use the convention `hideInformationForXXX`.

Reply:
Cool, since the setting of attributes is tied to the `logic` attribute, I've made everything explicit